src/ferc_xbrl_extractor/instance.py

```diff
@@ -251,45 +231,36 @@ def __init__(
 
         Args:
             contexts: Dictionary mapping context ID to contexts.
-            facts: Dictionary mapping context ID to facts.
+            instant_facts: Dictionary mapping concept name to list of instant facts.
+            duration_facts: Dictionary mapping concept name to list of duration facts.
             filing_name: Name of parsed filing.
         """
         # This is a nested dictionary of dictionaries to locate facts by context
-        self.instant_facts: FactDict = {}
-        self.duration_facts: FactDict = {}
-
+        self.instant_facts = instant_facts
+        self.duration_facts = duration_facts
         self.filing_name = filing_name
+        self.contexts = contexts
 
-        for c_id, context in contexts.items():
-            axes = tuple(
-                sorted(
-                    stringcase.snakecase(dim.name) for dim in context.entity.dimensions
-                )
-            )
-            if context.period.instant:
-                if axes not in self.instant_facts:
-                    self.instant_facts[axes] = {}
-
-                self.instant_facts[axes][context] = facts[c_id]
-            else:
-                if axes not in self.duration_facts:
-                    self.duration_facts[axes] = {}
-
-                self.duration_facts[axes][context] = facts[c_id]
-
-    def get_facts(self, instant: bool, axes: list[str]) -> dict[Context, list[Fact]]:
-        """Return a dictionary that maps Contexts to a list of facts for each context.
+    def get_facts(
+        self, instant: bool, concept_name: str, primary_key: list[str]
+    ) -> dict[str, list[Fact]]:
+        """Return a dictionary that maps Context ID's to a list of facts for each context.
 
         Args:
             instant: Get facts with instant or duration period.
-            axes: List of axis names defined for fact table.
+            concept_name: Name of context which maps to a column name and name of facts.
+            primary_key: Name of columns in primary_key used to filter facts.
         """
         if instant:
-            facts = self.instant_facts.get(tuple(sorted(axes)), {})
+            period_fact_dict = self.instant_facts
         else:
-            facts = self.duration_facts.get(tuple(sorted(axes)), {})
+            period_fact_dict = self.duration_facts
 
-        return facts
+        return {
+            fact.c_id: fact
+            for fact in period_fact_dict.get(concept_name, [])
+            if self.contexts[fact.c_id].check_dimensions(primary_key)
```

Comment:
If we are joining this to the filtered contexts later, we might not need to filter by "is the context applicable to the table?" here.